base/distributed/managers.jl

```diff
@@ -455,10 +455,10 @@ end
 const client_port = Ref{Cushort}(0)
 
 function socket_reuse_port()
-    @static if is_linux() || is_apple()
+    if Bool(ccall(:jl_has_so_reuseport, Int32, ()))
```

Comment:
Also https://github.com/JuliaLang/julia/pull/22303/files#diff-5b6252dbf4f6ba35c7ee92dde3cb6451L836 can return `-1` now as Julia code is expected to check with `jl_has_so_reuseport`  before calling `jl_tcp_reuseport`